src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('All sides of the triangle must be greater than zero');
+    }
+
+    const longestSideTriangle = Math.max(this.a, this.b, this.c);
+
+    if (longestSideTriangle >= this.a + this.b
+      || longestSideTriangle >= this.b + this.c
+      || longestSideTriangle >= this.c + this.a
+    ) {
+      throw new Error('Longest side of triangle should be less '
+        + 'than the sum of the other two sides');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(p * (p - this.a)
+      * (p - this.b) * (p - this.c)) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('The radius of the circle must be greater than zero');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * (this.radius ** 2)) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (this.width < 0 || this.height < 0) {
+      throw new Error('Width and height of the rectangle must be non-negative');
+    }
+  }
+
+  getArea(): number {
+    return this.width * this.height;
+  }
```

Comment:
The area calculation should be rounded down to hundredths using Math.floor(area * 100) / 100.
```suggestion
  getArea(): number {
    const area = this.width * this.height;
    return Math.round(area * 100) / 100;
  }
```